src/build/rewrite-webcomponents-loader.ts

```diff
@@ -54,16 +82,57 @@ export class UseES5WebcomponentsLoader extends stream.Transform {
       return;
     }
     const parsed = parse5.parse(contents);
-    const script = dom5.nodeWalk(parsed, scriptIncludeWebcomponentsLoader);
+    const script = dom5.nodeWalk(parsed, webcomponentsLoaderMatcher);
     if (!script) {
       callback(null, file);
       return;
     }
 
-    const scriptPath = dom5.getAttribute(script, 'src')!;
-    const scriptUrl = url.resolve(scriptPath, 'custom-elements-es5-adapter.js');
-    const es5AdapterScript = parse5.parseFragment(`<script src="${scriptUrl}"></script>`);
-    dom5.insertBefore(script.parentNode!, script, es5AdapterScript);
+    // Collect important dom references & create fragments for injection
+    const bodyElement = dom5.query(parsed, bodyMatcher)!;
+    const headElement = dom5.query(parsed, headMatcher)!;
+    const loaderScriptUrl = dom5.getAttribute(script, 'src')!;
+    const adapterScriptUrl =
+        url.resolve(loaderScriptUrl, 'custom-elements-es5-adapter.js');
+    const loaderMovedComment = dom5.constructors.comment(
+        ' AUTO-GENERATED BY JS-COMPILE: "webcomponents-loader.js" and ' +
+        'related resources have been moved to <body> so that ' +
+        '"custom-elements-es5-adapter.js" polyfill could be properly ' +
+        'injected into this file. ');
+    const es5AdapterScript = parse5.parseFragment(`
+<div id="autogenerated-ce-es5-shim">
+  <script type="text/javascript">
+    // This prevents custom-elements-es5-adapter.js from parsing & running
+    // on browsers without native support for ES2015+ & custom elements.
+    if (!window.customElements) {
+      var ceShimContainer = document.querySelector('#autogenerated-ce-es5-shim');
+      ceShimContainer.parentElement.removeChild(ceShimContainer);
+    }
+  </script>
+  <script type="text/javascript" src="${adapterScriptUrl}"></script>
+</div>
+`);
+
+    // Get references to all relevent siblings following the "loader" script
+    const loaderScriptSiblings = dom5.queryAll(
+        script.parentNode!,
+        dom5.predicates.OR(scriptMatcher, styleMatcher, linkMatcher));
+    const siblingsAfterLoader =
+        loaderScriptSiblings.splice(loaderScriptSiblings.indexOf(script));
+    dom5.insertBefore(headElement, script, loaderMovedComment);
```

Comment:
Here we assume the loader is a child of `<head>`, but that might not be the case. If the loader is already a child of `<body>` it should be enough to just insert the es5 adapter before it, w/o moving the siblings after it